flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/runner/DefaultDispatcherRunnerITCase.java

```diff
@@ -262,7 +272,17 @@ private DispatcherRunner createDispatcherRunner() throws Exception {
         return dispatcherRunnerFactory.createDispatcherRunner(
                 dispatcherLeaderElectionService,
                 fatalErrorHandler,
-                () -> jobGraphStore,
+                new JobPersistenceComponentFactory() {
```

Comment:
Should we have a testing implementation that allows you to pass the stores via constructor instead? (similar to `TestingCheckpointRecoveryFactory`)